zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/tenant/DbTenantStateTest.java

```diff
@@ -79,4 +79,90 @@ void shouldNotFindTenantByNonExistingId() {
     // then
     assertThat(tenantKey).isEmpty();
   }
+
+  @Test
+  void shouldUpdateTenantName() {
+    // given
+    final long tenantKey = 1L;
+    final String tenantId = "tenant-1";
+    final String oldName = "Tenant One";
+    final String newName = "Updated Tenant";
+    final var tenantRecord =
+        new TenantRecord().setTenantKey(tenantKey).setTenantId(tenantId).setName(oldName);
+
+    tenantState.createTenant(tenantRecord);
+
+    // when
+    final var updatedRecord =
+        new TenantRecord().setTenantKey(tenantKey).setTenantId(tenantId).setName(newName);
+    tenantState.updateTenant(updatedRecord);
+
+    // then
+    final var persistedTenant = tenantState.getTenantByKey(tenantKey);
+    assertThat(persistedTenant).isPresent();
+    assertThat(persistedTenant.get().getName()).isEqualTo(newName);
+  }
+
+  @Test
+  void shouldUpdateTenantId() {
+    // given
+    final long tenantKey = 1L;
+    final String oldTenantId = "tenant-1";
+    final String newTenantId = "tenant-2";
+    final String tenantName = "Tenant One";
+    final var tenantRecord =
+        new TenantRecord().setTenantKey(tenantKey).setTenantId(oldTenantId).setName(tenantName);
+
+    tenantState.createTenant(tenantRecord);
+
+    // when
+    final var updatedRecord =
+        new TenantRecord().setTenantKey(tenantKey).setTenantId(newTenantId).setName(tenantName);
+    tenantState.updateTenant(updatedRecord);
+
+    // then
+    // Old tenant ID should not be found
+    final var oldKey = tenantState.getTenantKeyById(oldTenantId);
+    assertThat(oldKey).isEmpty();
+
+    // New tenant ID should be mapped to the tenantKey
+    final var newKey = tenantState.getTenantKeyById(newTenantId);
+    assertThat(newKey).isPresent();
+    assertThat(newKey.get()).isEqualTo(tenantKey);
+  }
+
+  @Test
+  void shouldUpdateTenantNameAndEntityKey() {
+    // given
+    final long tenantKey = 1L;
+    final String tenantId = "tenant-1";
+    final String oldName = "Tenant One";
+    final String newName = "Updated Tenant";
+    final long oldEntityKey = 123L;
+    final long newEntityKey = 456L;
+    final var tenantRecord =
+        new TenantRecord()
+            .setTenantKey(tenantKey)
+            .setTenantId(tenantId)
+            .setName(oldName)
+            .setEntityKey(oldEntityKey);
```

Comment:
❌ We don't really want to store the `entityKey` in the state. We should change the `PersistedTenant` similar to what we did with the `PersistedRole`. This means instead of having a single `ObjectProperty` on there we should remove that and add a `tenantKeyProperty`, `tenantIdProperty` and `tenantNameProperty`